repos/system_upgrade/el7toel8/actors/repositoriesexcluded/libraries/repositoriesexcluded.py

```diff
@@ -0,0 +1,110 @@
+from leapp import reporting
+from leapp.libraries.common.config import get_product_type
+from leapp.libraries.stdlib import api
+from leapp.models import (
+    CustomTargetRepository,
+    RepositoriesExcluded,
+    RepositoriesFacts,
+    RepositoriesMap,
+)
+
+already_reported_skips = []
+
+
+def _is_optional_repo(repo, custom_target_repos):
+    global already_reported_skips  # pylint: disable-msg=global-statement
+    sys_type = get_product_type("source")
+    suffix = "optional-rpms"
+    if sys_type != "ga":
+        suffix = "optional-{}-rpms".format(sys_type)
+    for enabled_repo in custom_target_repos:
+        if enabled_repo.repoid == repo.to_repoid:
+            if repo.to_repoid not in already_reported_skips:
+                already_reported_skips.append(repo.to_repoid)
+                api.current_logger().info(
+                    "Skipping %s repository from exclusion due to directly "
+                    "specified in --enablerepo flag"
+                )
+                report = [
+                    reporting.Title(
+                        "Added not supported by Red Hat repository"
+                    ),
+                    reporting.Summary(
+                        "The following repository added to the upgrade, "
+                        "but it is not supported by the Red Hat.:\n- {}".format(
+                            enabled_repo.repoid
+                        )
+                    ),
+                    reporting.Severity(reporting.Severity.MEDIUM),
+                    reporting.Tags([reporting.Tags.REPOSITORY]),
+                ]
+                reporting.create_report(report)
```

Comment:
Frankly, I don't like adding this whole reporting feature to the `_is_optional_repo()` function because then the name of the function does not fit what it is doing.

Please create a new function like:
```
def _report_using_unsupported_repo(opt_repo_mapping):
    custom_target_repos = tuple(api.consume(CustomTargetRepository))
    for repo_to_enable in custom_target_repos:
        if repo.to_repoid in opt_repo_mapping.values():
            <create_report>
```
and call it from the `process()`, like:
```
opt_repo_mapping = _get_optional_repo_mapping() # See my other comment below on this function
_report_using_unsupported_repo(opt_repo_mapping)
```

Reply:
Agree, sounds better